earthpy/tests/test_spatial.py

```diff
@@ -1,6 +1,79 @@
 from earthpy import spatial
+from osgeo import gdal
+from osgeo import osr
+import rasterio as rio
+import numpy as np
+import os
 
+# A helper function to write a 3D array of data to disk as a GeoTIFF 
+# The result will have no spatial info or relevance
+def dummy_tif_writer(arr, destfile):
+    """Writes a tif file to a specified location using an array.
+
+    Parameters
+    ----------
+    arr : numpy ndarray 
+        The array should have 3 dimensions, arranged such that
+        a the result of arr.shape is in the form [rows, columns, channels].
+        
+    destfile : filepath string
+        The filepath for where the GeoTIFF file will be written.
+    """    
+    
+
+    # Make sure the array is 3 dimensional, assuming last dimension is number of bands
+    assert len(arr.shape) == 3
+    
+    try:
+        
+        dummy_geotrans = (0,1,0,0,0,-1)
+
+        # Get the dimensions of the array
+        y_pixels, x_pixels, n_channels = arr.shape
+
+        # Create the GeoTIFF file
+        driver = gdal.GetDriverByName('GTiff')
+        dataset = driver.Create(dst_filename, x_pixels, y_pixels, int(n_channels), gdal.GDT_Float32)
+
+        # Write the bands to the GeoTIFF
+        for i in range(n_channels):
+            dataset.GetRasterBand(i+1).WriteArray(dummy[:,:,i])
+
+        # Define the Unknown projection
+        srs = osr.SpatialReference()            
+        srs.ImportFromEPSG(0)
+        proj = srs.ExportToWkt() 
+
+        # Set the spatial properties of the GeoTIFF file
+        dataset.SetGeoTransform(geotrans)
+        dataset.SetProjection(proj)
+
+        dataset.FlushCache()
+
+        # Remove the dataset from memory
+        dataset = None
+    
+        return (0, dst_filename)
+    
+    except Exception as e:
+        
+        return(e, None)
+    
 
 def test_dummy():
     """"Dummy test that will pass."""
     assert 1 == 1
+    
+def test_dummy_tif_writer():
+    """ Testing dummy_tif_writer."""
+    
+    destfile = 'dummy.tif'
+    arr = np.ones((5,5,1))
+    code,fi = dummy_tif_writer(arr, destfile)
+    
+    assert code == 0
+    assert os.path.exists(destfile) is True
+    
+    # Clean up the file
+    if os.path.exists(destfile):
+        os.remove(destfile)
```

Comment:
@mbjoseph it looks like there are tests here. Are there other items you'd like to see in this PR or can we merge it?

Reply:
This test looks good to make sure it works how we expect it to, but I'd love to see one more test to make sure it *doesn't* work how we expect it to not work, i.e., a test that hits the `except` part of the `try...except...` logic of the function. @joemcglinchy can you add a test case that returns the exception that you implemented at the end of `dummy_tif_writer` (or remove the exception if it's not needed)?